test/SAMtags/README.md

```diff
@@ -0,0 +1,35 @@
+MM and ML auxiliary tags
+========================
+
+The purpose of these test files is to test parsing of the MM and ML
+tags.  These succint MM and ML tags are present in the .sam files,
+with a more human readable expanded form in the .txt files.
+Developers should check whether their implementation is able to
+convert between the two forms.
+
+The .sam files are SAM format, but the only fields used for this
+test are the reverse-complementation flag (FLAG bit 0x10), the
+sequence, and the MM and ML tags.
+
+The .txt files uses one line per base, with a blank line separating
+sequeces.  Each line consists of two table separated fields
+representing the top (original as-sequenced orientation) and bottom
+strand.
+
+Each field in the .txt files is a concatenation of the canonical base
+and any modifications with their associated probability, expressed
+here as a rounded-down percentage.  The modification and percentage
+are joined together with no punctuation, but for ChEBI numeric codes
+the code is bracketted to distinguish it from the numeric percentage.
+For example "Cm80(76792)73" is canonical base C, modification m (80%)
+and modification ChEBI 76792 (73%).
+
+
+The parse_mm.pl perl script can convert .sam to .txt files, but is for
```

Comment:
why not add this to the test suite? we already have a make file...

Reply:
We could do, but I'm not convinced this is what the test suite is for given this isn't really an implementation or a usable program.   I guess it could spot mistakes if we ever add additional MM test files.

I'm more interesting in *human* validation that the parsing of these tests make sense, given I basically produced the expected .txt output by running that perl script in the first place. ;-)

I did go through them myself checking, but another set of eyes would be good.

Reply:
Also, do we want to add a dependency on perl?  I guess as we have no `make check` or `make test` Makefile target then it doesn't matter and won't break anything.

I'll make it so...